include/boost/math/concepts/concepts.hpp

```diff
@@ -0,0 +1,111 @@
+//  (C) Copyright Matt Borland 2022.
+//  Use, modification and distribution are subject to the
+//  Boost Software License, Version 1.0. (See accompanying file
+//  LICENSE or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_MATH_CONCEPTS_CONCEPTS_HPP
+#define BOOST_MATH_CONCEPTS_CONCEPTS_HPP
+
+#if __cplusplus >= 202002L || _MSVC_LANG >= 202002L
+#if __has_include(<concepts>)
+#include <concepts>
+
+#include <type_traits>
+#include <boost/math/tools/config.hpp>
+
+namespace boost::math::concepts {
+
+template <typename T>
+concept Integral = std::is_integral_v<T>
+                   #ifdef __SIZEOF_INT128__
+                   || std::is_same_v<__int128_t, T>
```

Comment:
Would this work with `gmp` integral types?

Reply:
Should built-ins or arbitrary precision types be the special case? I am thinking `BOOST_MATH_INTEGRAL` and `BOOST_MATH_ARBITRARY_INTEGRAL`